### PR TITLE
Created ComposeAWSutilsString to organize steps 

### PR DESCRIPTION
**Pull request by oblandford-novetta**
Changes offered/proposed in this pull request:
- Created function ComposeCliOptionString
- ComposeAWSutilsString
- Cosmetic changes and formatting

* New PR Alert to: @plus3it/spel

### CODE REVIEW

**Comment by oblandford-novetta** [comment]
go codebuild go

File: spel/scripts/amigen7-build.sh [hunk 1]
```diff
@@ -220,6 +219,54 @@ function CollectManifest {
...
+
+    # Whether to install AWS CLIv2
+    if [[ -n "${AWSCLIV2SOURCE}" ]]
+    then
+      CLIOPT_AWSUTILS+=("-c ${AWSCLIV2SOURCE}")
+    fi
+
+    # Epel mirrors are maddening; retry 5 times to work around issues
+    echo "Executing AWSutils.sh"
+    retry 5 bash -eux -o pipefail "${ELBUILD}/AWSutils.sh ${CLIOPT_AWSUTILS[*]}"
```
Comment (lorengordon) [inline spel/scripts/amigen7-build.sh:240]: Similar to the other function, `ComposeCliOptionString`, will it work if we leave the three lines above in the "main" part of the workflow? This function `ComposeAWSutilsString` would literally just set the variables, and calling `AWSutils.sh` would happen outside the function...

Comment (oblandford-novetta) [reply spel/scripts/amigen7-build.sh]: Moved the lines to call the AWSutils.sh script back into main and out of the function that is meant to set the variables.

File: spel/scripts/amigen7-build.sh [hunk 2]
```diff
@@ -220,6 +219,78 @@ function CollectManifest {
...
+    
+    # Construct the cli option string for a custom repo
+    CLIOPT_CUSTOMREPO=""
+    if [[ -n "${CUSTOMREPORPM}" && -n "${CUSTOMREPONAME}" ]]
+    then
+        CLIOPT_CUSTOMREPO=(-r "${CUSTOMREPORPM}" -b "${CUSTOMREPONAME}")
+    fi
+}
+
+function ComposeCliOptionString {
```
Comment (lorengordon) [inline spel/scripts/amigen7-build.sh:267]: This function is now present twice, causing shellcheck to freak out (technical term)

**Comment by oblandford-novetta** [comment]
go codebuild go

**Comment by oblandford-novetta** [comment]
go codebuild go

**Comment by oblandford-novetta** [comment]
go codebuild go

**Review by lorengordon — APPROVED** [review]
Works for me 🚢

**Pull request merged**